server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SievePoster.java

```diff
@@ -20,36 +20,34 @@
 package org.apache.james.transport.mailets.jsieve.delivery;
 
 import javax.mail.MessagingException;
-import javax.mail.internet.MimeMessage;
 
-import org.apache.james.transport.mailets.delivery.MailboxAppender;
+import org.apache.james.transport.mailets.delivery.MailStore;
 import org.apache.james.transport.mailets.jsieve.Poster;
 import org.apache.james.user.api.UsersRepository;
 import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.mailet.Mail;
 import org.apache.mailet.MailAddress;
 
 public class SievePoster implements Poster {
-
-    private final MailboxAppender mailboxAppender;
```

Comment:
Appends are done by LocalDelivery.

Targetted folder is passed threw mail attributes.

Note : Such mechanism might allow other mecanism to do filtering in mailets without any difficulty.